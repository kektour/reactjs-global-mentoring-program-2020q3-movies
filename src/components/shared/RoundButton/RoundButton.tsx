import React from 'react';
import classnames from 'classnames';
import styles from './RoundButton.module.scss';

type Props = {
  classes?: {
    root?: string;
  };
  onClick: (e: React.MouseEvent<HTMLDivElement>) => void;
};

export const RoundButton: React.FC<Props> = (props) => {
  const { classes = {}, onClick: handleClick } = props;
  return (
    <div className={classnames(styles.root, classes.root)} onClick={handleClick}>
      <div className={styles.dot}></div>
      <div className={styles.dot}></div>
      <div className={styles.dot}></div>
    </div>
  );
};
