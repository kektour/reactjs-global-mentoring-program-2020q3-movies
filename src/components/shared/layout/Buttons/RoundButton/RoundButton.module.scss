.root {
  width: 40px;
  height: 40px;
  background: #232323;
  border-radius: 50%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.dot {
  width: 4px;
  height: 4px;
  background-color: #fff;
  border-radius: 50%;
  margin: 2px;
}
