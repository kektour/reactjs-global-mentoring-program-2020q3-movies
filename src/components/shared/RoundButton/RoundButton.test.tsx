import React from 'react';
import { render } from '@testing-library/react';
import { RoundButton } from './RoundButton';

describe('RoundButton', () => {
  it('Should render component correctly', () => {
    const component = render(<RoundButton onClick={() => null} />);
    expect(component.asFragment()).toMatchSnapshot();
  });
});
