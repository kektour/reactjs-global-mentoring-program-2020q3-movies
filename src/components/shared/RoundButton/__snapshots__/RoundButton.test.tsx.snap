// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundButton Should render component correctly 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="dot"
    />
    <div
      class="dot"
    />
    <div
      class="dot"
    />
  </div>
</DocumentFragment>
`;
